test/org/loklak/api/search/InstagramProfileScraperTest.java

```diff
@@ -0,0 +1,42 @@
+package org.loklak.api.search;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import org.junit.Test;
+import org.loklak.api.search.InstagramProfileScraper;
+import org.loklak.data.DAO;
+import org.loklak.http.ClientConnection;
+import org.json.JSONArray;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * These unit-tests test org.loklak.api.search.InstagramProfileScraper.java
+ */
+
+public class InstagramProfileScraperTest {
+
+	@Test
+	public void instagramProfileScraperUserTest() {
+
+		InstagramProfileScraper instagramScraper = new InstagramProfileScraper();
+		String url = "https://www.instagram.com/fossasia/";
+		BufferedReader br = null;
+
+		try {
+			ClientConnection connection = new ClientConnection(url);
+			//Check Network issue
+			assertThat(connection.getStatusCode(), is(200));
+			System.out.println("saurabh well done connected");
+			br = instagramScraper.getHtml(connection);
+		} catch (IOException e) {
+			DAO.log("InstagramProfileScraperTest.instagramProfileScraperUserTest() failed to connect to network. url:" + url);
+		}
+
+		JSONArray instaProfile = new JSONArray();
+		instaProfile = instagramScraper.scrapeInstagram(br, url);
+		assertNotNull(instaProfile);
```

Comment:
You need to add more checks on the elements scraped like profile pic URL, bio, etc. At least check that they exist.